graphql-lwt/src/graphql_lwt.ml

```diff
@@ -21,11 +21,27 @@ module Server = struct
     Schema.execute schema ctx ~variables doc
 
   let execute_request ctx schema req body =
-    Cohttp_lwt.Body.to_string body >>= fun body' ->
-    Lwt_io.printf "Body: %s\n" body';
-    let json = Yojson.Basic.from_string body' in
-    let query = Yojson.Basic.(json |> Util.member "query" |> Util.to_string) in
-    let variables = try Yojson.Basic.Util.(json |> member "variables" |> to_assoc) with _ -> [] in
+    let uri = Cohttp.Request.uri req in
+    (* Use query params if the "query" query param exists; otherwise use the
+       body of the request *)
+    (match Uri.get_query_param uri "query" with
+      | Some query ->
+        Cohttp_lwt.Body.drain_body body >|= fun () ->
+        let variables =
+          match Uri.get_query_param uri "variables" with
+          | Some json ->
+            Yojson.Basic.(from_string json |> Util.to_assoc)
+          | None -> []
+        in
+        query, variables
+      | None ->
+        Cohttp_lwt.Body.to_string body >|= fun body' ->
+        Lwt_io.printf "Body: %s\n" body';
+        let json = Yojson.Basic.from_string body' in
+        let query = Yojson.Basic.(json |> Util.member "query" |> Util.to_string) in
+        let variables = try Yojson.Basic.Util.(json |> member "variables" |> to_assoc) with _ -> [] in
+        query, variables)
```

Comment:
Maybe this `match` could be extracted to a helper function `Uri.t -> Cohttp_lwt.Body.t -> string * Yojson.Basic.json`?